src/lightning/pytorch/trainer/trainer.py

```diff
@@ -965,33 +949,16 @@ def _teardown(self) -> None:
         self._signal_connector.teardown()
 
     def _run_stage(self) -> Optional[Union[_PREDICT_OUTPUT, _EVALUATE_OUTPUT]]:
-        # wait for all to join if on distributed
-        self.strategy.barrier("run-stage")
-
         if self.evaluating:
-            with self.profiler.profile(f"run_{self.state.stage}_evaluation"):
-                eval_loop_results = self._evaluation_loop.run()
-            # remove the tensors from the eval results
-            return convert_tensors_to_scalars(eval_loop_results)
```

Comment:
Moved to the `_impl` methods that use it

Reply:
How about moving it to `evaluation_loop.run()`?

Reply:
Fit's validation doesn't need it